worlds/doom_1993/__init__.py

```diff
@@ -0,0 +1,296 @@
+                continue
+            if loc["index"] == -1:
+                loc_id = None
+            self.create_location(loc_id, loc, self.multiworld.get_region(loc["region"], self.player))
+            self.location_count += 1
+        
+        # Create exits
+        self.multiworld.get_region("Menu", self.player).exits.append(Entrance(self.player, "Menu -> Mars", self.multiworld.get_region("Menu", self.player)))
+        self.multiworld.get_entrance("Menu -> Mars", self.player).connect(self.multiworld.get_region("Mars", self.player))
+
+        # Rest of regions
+        for region_name in Regions.regions:
+            self.create_2way_exit("Mars", region_name)
+
+
+    def create_2way_exit(self, region1_name, region2_name):
+        region1 = self.multiworld.get_region(region1_name, self.player)
+        region2 = self.multiworld.get_region(region2_name, self.player)
+
+        entrance1 = Entrance(self.player, region1_name + " -> " + region2_name, region1)
+        entrance2 = Entrance(self.player, region2_name + " -> " + region1_name, region2)
+
+        region1.exits.append(entrance1)
+        region2.exits.append(entrance2)
+
+        entrance1.connect(region2)
+        entrance2.connect(region1)
+
+
+    def completion_rule(self, state):
+        for event in Events.events:
+            if not (event in self.location_name_to_id):
+                continue
+            loc = Locations.location_table[self.location_name_to_id[event]]
+            if not self.included_episodes[loc["episode"] - 1]:
+                continue
+            if not state.has(event, self.player, 1):
+                return False
+        return True
+
+
+    def set_rules(self):
+        Rules.set_rules(self)
+        self.multiworld.completion_condition[self.player] = lambda state: self.completion_rule(state)
+    
+    
+    def create_item(self, name: str) -> DOOM1993Item:
+        item_id: int = self.item_name_to_id[name]
+        doom_1993_item = DOOM1993Item(name, Items.item_table[item_id]["classification"], item_id, player=self.player)
+        return doom_1993_item
+    
+    
+    def create_item_with_classification(self, name: str, classification: ItemClassification) -> DOOM1993Item:
+        item_id: int = self.item_name_to_id[name]
+        doom_1993_item = DOOM1993Item(name, classification, item_id, player=self.player)
+        return doom_1993_item
+    
+
+    def place_locked_item_in_locations(self, item_name, locations):
+        location = self.multiworld.random.choice(locations)
+        self.multiworld.get_location(location, self.player).place_locked_item(self.create_item(item_name))
+        self.location_count -= 1
+
+
+    def create_items(self):
+        is_only_first_episode = self.get_episode_count() == 1 and self.included_episodes[0]
+
+        # Items
+        for item_id, item in Items.item_table.items():
+            if item["name"] == "Shotgun":
+                tmp = 5
```

Comment:
This doesn't seem to have a purpose. Or if it does, it's unclear.

Reply:
Temp breakpoint debug line :)
Will remove